src/vos/vos_obj_cache.c

```diff
@@ -230,12 +230,23 @@ vos_obj_hold(struct daos_lru_cache *occ, daos_handle_t coh,
 
 		if ((!(obj->obj_df->vo_oi_attr & VOS_OI_PUNCHED) ||
 		     obj->obj_df->vo_latest >= epoch) &&
-		    obj->obj_df->vo_incarnation == obj->obj_incarnation) {
-			if (obj->obj_epoch <= epoch)
-				goto out; /* belong to the same incarnation */
+		    (obj->obj_df->vo_incarnation == obj->obj_incarnation) &&
+		    (obj->obj_epoch <= epoch || obj->obj_incarnation == 0)) {
+			struct umem_instance	*umm = &cont->vc_pool->vp_umm;
+
+			rc = vos_dtx_check_availability(umm, coh,
+						obj->obj_df->vo_dtx,
+						umem_ptr2off(umm, obj->obj_df),
+						intent, DTX_RT_OBJ);
+			if (rc < 0) {
+				vos_obj_release(occ, obj);
+				D_GOTO(failed, rc);
+			}
```

Comment:
Check dtx will access pmem directly? hmm, this is object cache, suppose to be in DRAM.  I think pmem read latency is still pretty high.

"On average, the average idle read latency with Optane persistent memory is expected to be about 350 nanoseconds when applications direct the read operation to Optane persistent memory, or when the requested data is not cached in DRAM."

I believe DRAM is less than 20 nanoseconds.

Reply:
It is not easy to avoid the PMEM accessing for availability check because we have to handle shared update case unless we also cache DTX related logic, that will make things more complex. I prefer to resolve performance related things in next step.